artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/AddressControlImpl.java

```diff
@@ -373,6 +372,86 @@ public String sendMessage(final Map<String, String> headers,
       return MBeanInfoHelper.getMBeanAttributesInfo(AddressControl.class);
    }
 
+   @Override
+   public void pause() throws Exception {
+      pause(false);
+   }
+
+   @Override
+   public void pause(boolean persist) throws Exception {
+      if (AuditLogger.isEnabled()) {
+         AuditLogger.pause(addressInfo);
+      }
+      checkStarted();
+
+      clearIO();
+      try {
+         if (!addressInfo.isPaused()) {
+            addressInfo.pause();
+            if (persist) {
+               server.getStorageManager().storeAddressStatus(addressInfo.getId(), AddressStatus.PAUSED);
+            }
```

Comment:
I think this logic should be moved into the `AddressInfo#pause` method so users of `AddressInfo` get the persistence functionality without having to use the control object. Also, there needs to be a way to detect multiple invocations of `pause` so that PAUSED records don't just keep getting added over and over as that will cause the bindings journal size to increase over time.

Reply:
Moving the method to AddressInfo would make us introduce serious refactoring into Addressinfo
AddressNfo would on that case have a similar power to what Queue does. 

That is, we would need to bring the Server, StorageManager and other functionality that wouldn't belong to AddressInfo, unless we make it similar to Queue.. (we could even rename the class to Address)...

Why don't we keep this on AddressController for now.. with a // TODO block to move this to AddressInfo, and make proper refactoring on the Address as a separate task?

We would need to at least change the contructor to contain the Server (you can find everything else after the ACtiveMQServer).

Reply:
A simple solution would be to create a setter on Addressinfo, and we call the setter right before we call the pause method.

We could throw an IllegalSTateException if the server is not informed. So the API would be... setServer(server);addressInfo.pause(true/false);

Reply:
ok